src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -87,4 +97,34 @@ public int getTotalMessageCount() {
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
+  
+  /** Returns the longest message length of all users. */
+  public int getLongestMessageCount() {
+    return longestMessage;
+  }
+  
+  /** Returns the total number of users that have posted. */
+  public int getTotalUserCount() {
+    return users.size();
+  }
+  
+  /** Returns the top three users that have posted on the website. */
+  public ArrayList<String> getTopUsers() {
+    ArrayList<String> topUsers = new ArrayList<String>(3);
+    int numTopUsers = 3;
+    String currTopUser = "";
+
+    while (numTopUsers >= 0) {
```

Comment:
Any reason you used a while loop instead of a for loop here? Seems the latter would be more clear